### PR TITLE
Implement the extended CONNECT protocol from RFC 8441

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

**Comment by nox** [comment]
I could make `Protocol` own a `HeaderValue` instead of a `BytesStr`, but I don't really see the point of doing that given all other pseudo-headers are represented with a `BytesStr`, and making this change would also require a `set_pseudo!` refactor.

**Comment by nox** [comment]
I've decided to not implement new user error cases on the client side given there are none for normal connect requests either anyway.

File: src/ext.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+//! Extensions specific to the HTTP/2.0 protocol.
```
Comment (LPardue) [inline src/ext.rs:1]:
```suggestion
//! Extensions specific to the HTTP/2 protocol.
```

File: src/proto/settings.rs [hunk 2]
```diff
@@ -117,21 +117,22 @@ impl Settings {
...
             }
-
-            streams.apply_remote_settings(settings)?;
         }
 
         self.remote = None;
 
         match &self.local {
             Local::ToSend(settings) => {
+                tracing::trace!("AM TRYING: {:?}", settings);
```
Comment (LPardue) [inline src/proto/settings.rs:135]: is this intended to be left in?

File: src/server.rs [hunk 3]
```diff
@@ -1410,6 +1431,11 @@ impl proto::Peer for Peer {
             malformed!("malformed headers: missing method");
         }
 
+        let has_protocol = pseudo.protocol.is_some();
+        if !is_connect && has_protocol {
+            malformed!("malformed headers: missing method");
```
Comment (LPardue) [inline src/server.rs:1436]: I think you want a different message

**Review by seanmonstar — APPROVED** [review]
I think this looks great! Everything looks taken care of, all the situations have tests. Impressive!

**Comment by seanmonstar** [comment]
I'll leave merging up to you if there's anything you want to finish.

**Pull request merged**